bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/ReplicationWorker.java

```diff
@@ -186,9 +187,12 @@ private static void waitBackOffTime() {
 
     private void waitTillTargetBookieIsWritable() {
         LOG.info("Waiting for target bookie {} to be back in read/write mode", targetBookie);
-        while (admin.getReadOnlyBookies().contains(targetBookie)) {
+        while (workerRunning && admin.getReadOnlyBookies().contains(targetBookie)) {
+            isInReadOnlyMode = true;
             waitBackOffTime();
```

Comment:
I guess we could setup a watch to get a notification when the bookie comes back in rw mode, though  I don't think it's really critical.  The code above is already using a watch based cached set of read-only bookies, only checking that every 5secs. For replication worker there's no need to resume replicating within 5sec